packages/contracts-bedrock/test/DelayedVetoable.t.sol

```diff
@@ -0,0 +1,181 @@
+        assumeNonzeroData(data);
+
+        vm.prank(initiator);
+        (bool success,) = address(delayedVetoable).call(data);
+
+        vm.warp(block.timestamp + operatingDelay);
+        vm.expectEmit(true, false, false, true, address(delayedVetoable));
+        emit Forwarded(keccak256(data), data);
+
+        vm.expectCall({ callee: target, data: data });
+        (success,) = address(delayedVetoable).call(data);
+        assertTrue(success);
+    }
+}
+
+contract DelayedVetoable_HandleCall_TestFail is DelayedVetoable_Init {
+    /// @dev The delay is inititially set to zero and the call is immediately forwarded.
+    function test_handleCall_unauthorizedInitiation_reverts() external {
+        vm.expectRevert(abi.encodeWithSelector(Unauthorized.selector, initiator, address(this)));
+        (bool success,) = address(delayedVetoable).call(NON_ZERO_DATA);
+        assertTrue(success);
+    }
+
+    /// @dev The call cannot be forewarded until the delay has passed.
+    function testFuzz_handleCall_forwardingTooSoon_reverts(bytes memory data) external {
+        vm.prank(initiator);
+        (bool success,) = address(delayedVetoable).call(data);
+
+        vm.expectRevert(abi.encodeWithSelector(ForwardingEarly.selector));
+        (success,) = address(delayedVetoable).call(data);
+        assertFalse(success);
+    }
+
+    /// @dev The call cannot be forwarded a second time.
+    function testFuzz_handleCall_forwardingTwice_reverts(bytes memory data) external {
+        assumeNonzeroData(data);
+
+        // Initiate the call
+        vm.prank(initiator);
+        (bool success,) = address(delayedVetoable).call(data);
+
+        vm.warp(block.timestamp + operatingDelay);
+        vm.expectEmit(true, false, false, true, address(delayedVetoable));
+        emit Forwarded(keccak256(data), data);
+
+        // Forward the call
+        vm.expectCall({ callee: target, data: data });
+        (success,) = address(delayedVetoable).call(data);
+        assertTrue(success);
+
+        // Attempt to foward the same call again.
+        vm.expectRevert(abi.encodeWithSelector(Unauthorized.selector, initiator, address(this)));
+        (success,) = address(delayedVetoable).call(data);
+        assertTrue(success);
+    }
+
+    /// @dev If the target reverts, it is bubbled up.
+    function testFuzz_handleCall_forwardingTargetReverts_reverts(bytes memory inData, bytes memory outData) external {
+        assumeNonzeroData(inData);
+
+        // Initiate the call
+        vm.prank(initiator);
+        (bool success,) = address(delayedVetoable).call(inData);
+
+        vm.warp(block.timestamp + operatingDelay);
+        vm.expectEmit(true, false, false, true, address(delayedVetoable));
+        emit Forwarded(keccak256(inData), inData);
+
+        vm.mockCallRevert(target, inData, outData);
+
+        // Forward the call
+        vm.expectRevert(outData);
+        (bool success2,) = address(delayedVetoable).call(inData);
```

Comment:
Could you also assert eq on the `bytes memory` of the `call`?

Reply:
According to that same [gotcha](https://book.getfoundry.sh/cheatcodes/expect-revert?highlight=expectrevert#description) above: 

> Apart from this, `expectRevert` also mangles return data on low level calls, and is not usable.

This caused me much confusion

Reply:
also, `vm.expectRevert()` should be handling that assertion